apps/shelter-web/src/app/shared/clients/apollo/csrf.ts

```diff
@@ -0,0 +1,74 @@
+import { ApolloLink, FetchResult, Observable } from '@apollo/client';
+import { getItem, setItem } from '../../utils/storage/cookies';
+
+const CSRF_COOKIE_NAME = import.meta.env.VITE_CSRF_COOKIE_NAME;
+const CSRF_HEADER_NAME = import.meta.env.VITE_CSRF_HEADER_NAME;
+
+const csrfTokenRegex = new RegExp(`${CSRF_COOKIE_NAME}=([^;]+)`);
+
+const extractCsrfToken = async (apiUrl: string, customFetch = fetch) => {
+  let csrfToken = await getItem(CSRF_COOKIE_NAME);
+  if (!csrfToken) {
+    const response = await customFetch(apiUrl, { credentials: 'include' });
+    const cookies = response.headers?.get('Set-Cookie');
```

Comment:
This won't be available anyway in the FE, set-cookie I believe may be a protected header in the browser. I think this is only for mobile.